Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.16.9 as builder
+FROM golang:1.17.11 as builder
```

Comment:
Should this be a separate PR?

Reply:
i dont think so. it updates to match the version in CI. https://github.com/tinkerbell/pbnj/blob/34853cdbf069ad1efbfd4babba9fd0464e7c08ec/.github/workflows/ci.yaml#L17

Reply:
Will keep it in here. Thanks!